### PR TITLE
Add metrics and event docs

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add metrics and event docs

**Which issue(s) this PR fixes**:
Part of #258 

**Special notes for your reviewer**:
None

### CODE REVIEW

File: docs/reference/instrumentation/event.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: Karmada Event Reference
+---
+
+## Events
+
+This page details the event that different Karmada components export.
```
Comment (Tingtal) [inline docs/reference/instrumentation/event.md:7]:
```suggestion
This section details the events that different Karmada components export.
```

File: docs/reference/instrumentation/event.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+You can obtain the event by `kubectl describe` the corresponding object.
```
Comment (Tingtal) [inline docs/reference/instrumentation/event.md:8]:
```suggestion
You can obtain query events by running `kubectl describe` on the corresponding object.
```

File: docs/reference/instrumentation/metrics.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+title: Karmada Metrics Reference
+---
+
+This page details the metrics that different Karmada components export.
```
Comment (Tingtal) [inline docs/reference/instrumentation/metrics.md:5]:
```suggestion
This section details the metrics that different Karmada components export. 
```

File: docs/reference/instrumentation/metrics.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+title: Karmada Metrics Reference
+---
+
+This page details the metrics that different Karmada components export. 
+You can query the metrics endpoint for these components using an HTTP scrape, and fetch the current metrics data in Prometheus format.
+
+| Name                                       | Type      | Help                                                              | Labels | Source Components           |
+|--------------------------------------------|-----------|-------------------------------------------------------------------|--------|-----------------------------|
+| schedule_attempts_total                    | Counter   | Number of attempts to schedule resourceBinding.                   |        | karmada-scheduler           |
+| e2e_scheduling_duration_seconds            | Histogram | E2e scheduling latency in seconds.                                |        | karmada-scheduler           |
```
Comment (Tingtal) [inline docs/reference/instrumentation/metrics.md:11]:
```suggestion
| e2e_scheduling_duration_seconds            | Histogram | E2E scheduling latency in seconds.                                |        | karmada-scheduler           |
```

File: docs/reference/instrumentation/metrics.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+| scheduling_algorithm_duration_seconds      | Histogram | Scheduling algorithm latency in seconds(exclude scale scheduler). |        | karmada-scheduler           |
```
Comment (Tingtal) [inline docs/reference/instrumentation/metrics.md:12]:
```suggestion
| scheduling_algorithm_duration_seconds      | Histogram | Scheduling algorithm latency in seconds (excluding scale scheduler). |        | karmada-scheduler           |
```

**Review by Tingtal — CHANGES_REQUESTED** [review]
Small touches on wording. Pls check.

File: docs/reference/instrumentation/event.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+---
+
+## Events
+
+This section details the events which record key processes in Karmada.
+See more details about event [here](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/).
+
+| Reason                                  | Binding Objects                                                                             | Type    | Source Component                                                                                                          |
+|-----------------------------------------|---------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------------------------------|
+| CreateExecutionSpaceFailed              | cluster                                                                                     | Warning | cluster-controller                                                                                                        |
```
Comment (RainbowMango) [inline docs/reference/instrumentation/event.md:12]:
```suggestion
| CreateExecutionSpaceFailed              | Cluster                                                                                     | Warning | cluster-controller                                                                                                        |
```

File: docs/reference/instrumentation/event.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+| RemoveTargetClusterFailed               | cluster                                                                                     | Warning | cluster-controller                                                                                                        |
+| SyncImpersonationConfigFailed           | cluster                                                                                     | Warning | unified-auth-controller                                                                                                   |
+| SyncImpersonationConfigSucceed          | cluster                                                                                     | Normal  | unified-auth-controller                                                                                                   |
+| ReflectStatusFailed                     | work                                                                                        | Warning | work-status-controller                                                                                                    |
```
Comment (RainbowMango) [inline docs/reference/instrumentation/event.md:20]:
```suggestion
| ReflectStatusFailed                     | Work                                                                                        | Warning | work-status-controller                                                                                                    |
```

File: docs/reference/instrumentation/event.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+| InterpretHealthSucceed                  | work                                                                                        | Normal  | work-status-controller                                                                                                    |
+| SyncFailed                              | work                                                                                        | Warning | execution-controller                                                                                                      |
+| SyncSucceed                             | work                                                                                        | Normal  | execution-controller                                                                                                      |
+| CleanupWorkFailed                       | resourceBinding<br/>clusterResourceBinding                                                  | Warning | binding-controller<br/>cluster-resource-binding-controller                                                                |
```
Comment (RainbowMango) [inline docs/reference/instrumentation/event.md:26]:
```suggestion
| CleanupWorkFailed                       | ResourceBinding<br/>ClusterResourceBinding                                                  | Warning | binding-controller<br/>cluster-resource-binding-controller                                                                |
```

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**